test/framework/src/main/java/org/elasticsearch/transport/nio/SocketEventHandler.java

```diff
@@ -103,8 +104,8 @@ public void handleRead(NioSocketChannel channel) throws IOException {
      * @param channel that was being read
      * @param exception that occurred
      */
-    public void readException(NioSocketChannel channel, Exception exception) {
-        logger.trace("failed to read from channel", exception);
+    void readException(NioSocketChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("exception while reading from socket channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?